src/Common/src/Microsoft/Win32/SafeHandles/SafeX509Handles.Unix.cs

```diff
@@ -3,14 +3,24 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Diagnostics;
-using System.Security;
 using System.Runtime.InteropServices;
 
 namespace Microsoft.Win32.SafeHandles
 {
     internal sealed class SafeX509Handle : SafeHandle
     {
+#if DEBUG_FINALIZATIONS
```

Comment:
How bad would it be if this was DEBUG instead of DEBUG_FINALIZATIONS?

Reply:
For the X509Certificates tests setting the define changes the run from 10 seconds to 30 (on my test rig) due to callstack building.

So I'd be kind of opposed :)

Reply:
I'm curious, if you replaced:
```C#
private string _stackTrace = Environment.StackTrace;
```
with
```C#
private string _stackTrace = new StackTrace(fNeedFileInfo).ToString(System.Diagnostics.StackTrace.TraceFormat.Normal);
```
does it reduce the time to something reasonable?

Reply:
Nope.  But 

```C#
private StackTrace _stacktrace = new StackTrace(fNeedFileInfo: true);

~SafeX509Handle()
{
    Interop.Sys.PrintF("%s\n\n", $"{handle:X} {_stacktrace?.ToString() ?? "no stacktrace..."}");
}
```

(deferring the ToString() until it matters) reduces it to just under 13 seconds (whether capturing file info or not).

I think I'd also be concerned at how much noise it will introduce.  The Pkcs tests are less good at cleaning up their certificates, and currently cause 1255 finalizations (per grep/wc).  While it's probably good to get them down to 0 as well (to make sure that we're not inducing finalizations from that library's product code) it just seems distracting at the moment.

I could see `#if DEBUG` and then a runtime environment check?


```C#
private static readonly bool s_captureTrace =
    Environment.GetEnvironmentVariable("DEBUG_SAFEX509HANDLE_FINALIZATION") != null;

private StackTrace _stacktrace = s_captureTrace ? new StackTrace(fNeedFileInfo: true) : null;

~SafeX509Handle()
{
    if (s_captureTrace)
    {
        Interop.Sys.PrintF(
            "%s\n\n",
            $"0x{handle.ToInt64():x16} {_stacktrace?.ToString() ?? "no stacktrace..."}");
    }
}
```

?

Reply:
> I could see #if DEBUG and then a runtime environment check

Sounds like a good compromise, and eventually hopefully we could get to the point where we could remove the environment check.